account_commission/models/commission_settlement.py

```diff
@@ -69,52 +68,59 @@ def action_invoice(self):
-        move_form = Form(
-            self.env["account.move"].with_context(default_move_type="in_invoice")
-        )
-        if date:
-            move_form.invoice_date = date
         partner = self._get_invoice_partner()
-        move_form.partner_id = partner
-        move_form.journal_id = journal
+        vals = {
+            "move_type": "in_invoice",
+            "partner_id": partner.id,
+            "journal_id": journal.id,
+            "invoice_line_ids": [],
+        }
+        if date:
+            vals.update({"invoice_date": date})
         for settlement in self:
-            with move_form.invoice_line_ids.new() as line_form:
-                line_form.product_id = product
-                line_form.quantity = -1 if settlement.total < 0 else 1
-                line_form.price_unit = abs(settlement.total)
-                # Put period string
-                partner = self.agent_id
-                lang = self.env["res.lang"].search(
-                    [
-                        (
-                            "code",
-                            "=",
-                            partner.lang or self.env.context.get("lang", "en_US"),
-                        )
-                    ]
-                )
-                date_from = fields.Date.from_string(settlement.date_from)
-                date_to = fields.Date.from_string(settlement.date_to)
-                line_form.name += "\n" + _(
-                    "Period: from %(date_from)s to %(date_to)s",
-                    date_from=date_from.strftime(lang.date_format),
-                    date_to=date_to.strftime(lang.date_format),
+            # Put period string
+            lang = self.env["res.lang"].search(
+                [
+                    (
+                        "code",
+                        "=",
+                        self.agent_id.lang or self.env.context.get("lang", "en_US"),
+                    )
+                ]
+            )
+            date_from = fields.Date.from_string(settlement.date_from)
+            date_to = fields.Date.from_string(settlement.date_to)
+            vals["invoice_line_ids"].append(
+                (
+                    0,
+                    0,
+                    {
+                        "product_id": product.id,
+                        "quantity": -1 if settlement.total < 0 else 1,
+                        "price_unit": abs(settlement.total),
+                        "name": (
+                            "\n"
+                            + _(
+                                "Period: from %(date_from)s to %(date_to)s",
+                                date_from=date_from.strftime(lang.date_format),
+                                date_to=date_to.strftime(lang.date_format),
+                            ),
+                        ),
+                        # todo or compute agent currency_id?
+                        "currency_id": settlement.currency_id.id,
```

Comment:
I suppose we can remove these lines?  It's already a related field.

Reply:
wasn't sure what to do with these so I left it as is, but removing them should work

Reply:
I don't know in v16, but in previous versions, related fields are filled at the end of the pipeline, so if there's any reference to this on computations, it will be considered as empty, so it's not always a bad idea to preset it. This happens mostly on defaults based on related fields, but not sure if it can happen in other parts.